asdf/commands/edit.py

```diff
@@ -0,0 +1,529 @@
+            help="Output file (YAML for -e option, ASDF for -s option")
+
+        # The edit is either being performed or saved
+        group = parser.add_mutually_exclusive_group(required=True)
+        group.add_argument(
+            '-s',action='store_true',dest='save',
+            help="Saves a YAML text file to its ASDF file.  Requires an ASDF input file.")
+        group.add_argument(
+            '-e',action='store_true',dest='edit',
+            help="Create a YAML text file for a ASDF file.  Requires a YAML input file.")
+
+        parser.set_defaults(func=cls.run)
+
+        return parser
+
+    @classmethod
+    def run(cls, args):
+        """ Execute the edit subcommand.
+        """
+        return edit(args)
+
+
+def is_yaml_file ( fname ) :
+    '''
+    Determines if a file is a YAML file based only on the file extension.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    '''
+
+    base, ext = os.path.splitext(fname)
+    if '.yaml' != ext :
+        return False
+    return True
+
+
+def is_asdf_file ( fname ) :
+    '''
+    Determines if a file is ASDF based on file extension and the first
+    5 bytes of the file, which should be '#ASDF'.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    '''
+
+    base, ext = os.path.splitext(fname)
+    if '.asdf' != ext :
+        return False
+
+    with open(fname,"r+b") as fd :
+        first_string = "#ASDF"
+        first_line = fd.read(len(first_string)).decode('utf-8')
+        if first_string != first_line :
+            return False
+
+    return True
+
+
+def is_validate_path_and_ext ( fname, wanted_ext=None ) :
+    """ Validates the path exists and the extension is one wanted.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    wanted_ext : List of extensions to check.
+    """
+    if not os.path.exists(fname) :
+        print(f"Error: No file '{fname}' exists.")
+        return False
+
+    # Simply validates the path existence
+    if wanted_ext is None:  
+        return True
+    
+    # Make sure the extension is one desired.
+    base, ext = os.path.splitext(fname)
+    if ext not in wanted_ext:
+        return False
+
+    return True
+
+
+def is_validate_asdf_path ( fname ) :
+    """ Validates fname path exists and has extension '.asdf'.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    """
+    ext = ['.asdf']
+    if is_validate_path_and_ext(fname,ext) : 
+        return True
+    print(f"Error: '{fname}' should have extension '{ext[0]}'")
+    return False
+
+
+def is_validate_yaml_path ( fname ) :
+    """ Validates fname path exists and has extension '.yaml'.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    """
+    ext = ['.yaml']
+    if is_validate_path_and_ext(fname,ext) : 
+        return True
+    print(f"Error: '{fname}' should have extension '{ext[0]}'")
+    return False
+
+
+def validate_asdf_file ( fd ) :
```

Comment:
The name validate here is going to be somewhat confusing since validation for ASDF files has a specific meaning elsewhere. check_asdf_file? Perhaps someone else will have a better name suggestion.

Reply:
I changed "validate" to "check".